users/models.py

```diff
@@ -59,15 +81,40 @@ class User(AbstractUser):
         ),
         validators=[nickname_validator],
     )
+    bio = models.TextField(default="")
+    is_nfsw_visible = models.BooleanField(default=False)
+    is_followable = models.BooleanField(default=True)
+    is_content_visible = models.BooleanField(default=True)
+    is_communities_visible = models.BooleanField(default=True)
+
+    def __str__(self: "Profile") -> str:
+        return f"{self.nickname} settings"
+
+
+class User(AbstractUser):
+    MALE = "M"
+    FEMALE = "F"
+    NON_BINARY = "B"
+    OTHER = "O"
+    NON_SPECIFIED = "N"
+
+    GENDER_CHOICES = {
+        MALE: "Male",
+        FEMALE: "Female",
+        NON_BINARY: "Non-binary",
+        OTHER: "Other",
+        NON_SPECIFIED: "Non-specified",
+    }
+
     USERNAME_FIELD = "email"
-    REQUIRED_FIELDS: ClassVar[list[int]] = ["nickname"]
     objects = UserManager()
     username: None = None
     email: str = models.EmailField(unique=True)
+    gender = models.CharField(choices=GENDER_CHOICES, max_length=1)
     last_activity = models.DateTimeField(auto_now_add=True, db_index=True)
-
-    def __str__(self: "User") -> str:
-        return self.nickname
+    REQUIRED_FIELDS = ["gender"]
+    user_settings = models.OneToOneField(UserSettings, related_name="user", on_delete=models.SET_NULL, null=True)
```

Comment:
Why is it wrong?

Reply:
Why is `profile` ok, but not `user_settings`? I can't see a conflict.
```python
user_settings = models.OneToOneField(UserSettings, related_name="user", on_delete=models.SET_NULL, null=True
profile = models.OneToOneField(Profile, on_delete=models.SET_NULL, null=True, related_name="user")
```

Reply:
@jacoor what do you think?

Reply:
Clarity is really important. 
We could remove related_name no? How it would behave then?

Reply:
@yanazPL AI is wrong here. 
We do not need related_name on OneToOne. Please remove.
reference: https://akademiait.com.pl/zrozumiec-django-relacje-pomiedzy-modelami/ 

```
» ./manage.py shell
Python 3.10.7 (main, Sep 15 2022, 01:51:29) [Clang 14.0.0 (clang-1400.0.29.102)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> # import modeli
>>> from accounts.models import StudentCard, Student
>>> # pobranie i wyświetlenie pierwszego studenta
>>> student = Student.objects.first()
>>> student
<Student: John Doe>
>>> # pobranie pierwszej legitymacji
>>> student_card = StudentCard.objects.first()
>>> student_card
<StudentCard: John Doe 123456>
>>> # wyświetlenie studenta z poziomu legitymacji
>>> student_card.student
<Student: John Doe>
>>> student_card.student.first_name
'John'
>>> student_card.student.last_name
'Doe'
>>> # wyświetlenie danych legitymacji z poziomu studenta
>>> # czyli wykorzystanie relacji odwrotnej
>>> student.studentcard
<StudentCard: John Doe 123456>
>>> student.studentcard.card_id
'123456'
>>> student.studentcard.expiration_date
datetime.date(2023, 8, 17)
>>>
```

so it would simply be user.profile and profile.user. 
So please remove related name from profile and user_settings.